massgen/backend/llm_circuit_breaker.py

```
@@ -0,0 +1,481 @@
+                            agent_id=agent_id,
+                        )
+                        await asyncio.sleep(wait_seconds)
+                        continue
+
+                    # CAP -- no Retry-After, backoff + record failure
+                    self.record_failure(
+                        error_type="429_cap",
+                        error_message=str(exc)[:200],
+                    )
+                    if attempt < max_retries:
+                        jittered = delay * random.uniform(0.8, 1.2)
```

Ruff (0.15.6)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)